yt_dlp/extractor/reddit.py

```diff
@@ -147,19 +114,48 @@ def add_thumbnail(src):
                 for resolution in resolutions:
                     add_thumbnail(resolution)
 
-        return {
-            '_type': 'url_transparent',
-            'url': video_url,
+        info = {
+            'id': video_id,
             'title': data.get('title'),
             'thumbnails': thumbnails,
             'timestamp': float_or_none(data.get('created_utc')),
             'uploader': data.get('author'),
-            'duration': int_or_none(try_get(
-                data,
-                (lambda x: x['media']['reddit_video']['duration'],
-                 lambda x: x['secure_media']['reddit_video']['duration']))),
             'like_count': int_or_none(data.get('ups')),
             'dislike_count': int_or_none(data.get('downs')),
             'comment_count': int_or_none(data.get('num_comments')),
             'age_limit': age_limit,
         }
+
+        # Check if media is hosted on reddit:
+        reddit_video = try_get(data, (
+            lambda x: x[y]['reddit_video']
+            for y in ('media', 'secure_media')
+        ))
```

Comment:
btw, `traverse_obj` can be used to simplify this

or just
```suggestion
        reddit_video = traverse_obj(data, (('media', 'secure_media'), 'reddit_video'), get_all=False)
```

Reply:
🤯